### PR TITLE
ICU-21952 Add withoutLocale functions to LocalizedNumber[Range]Formatter

### PR DESCRIPTION
**Pull request by sffc**
Will send an API proposal shortly.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21952
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by sffc** [comment]
MSVC fails with:

```
            locale {
            FAIL: Formatter after resetLocale B: Unsafe Path; got "2 \u30E1\u30FC\u30C8\u30EB"; expected "2 \u00E3\u0192\u00A1\u00E3\u0192\u00BC\u00E3\u0192\u02C6\u00E3\u0192\u00AB"
            FAIL: Formatter after resetLocale B: Safe Path; got "2 \u30E1\u30FC\u30C8\u30EB"; expected "2 \u00E3\u0192\u00A1\u00E3\u0192\u00BC\u00E3\u0192\u02C6\u00E3\u0192\u00AB"
            FAIL: Formatter after resetLocale B: Skeleton Path: 'compact-long unit/meter unit-width-full-name': 2; got "2 \u30E1\u30FC\u30C8\u30EB"; expected "2 \u00E3\u0192\u00A1\u00E3\u0192\u00BC\u00E3\u0192\u02C6\u00E3\u0192\u00AB"
            FAIL: Formatter after resetLocale B: Concise Skeleton Path: 'compact-long unit/meter unit-width-full-name': 2; got "2 \u30E1\u30FC\u30C8\u30EB"; expected "2 \u00E3\u0192\u00A1\u00E3\u0192\u00BC\u00E3\u0192\u02C6\u00E3\u0192\u00AB"
         
            } ERRORS (4) in locale
```

I probably just need to fix the string encoding in the test file

**Review by richgillam — APPROVED** [review]
Looks great.

**Comment by sffc** [comment]
Need to change the function name to `withoutLocale`.

**Review by richgillam — APPROVED** [review]
Still LGTM.

**Comment by sffc** [comment]
I rebased this against main. Some files got moved around.

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/number/NumberRangeFormatterTest.java [hunk 1]
```diff
@@ -823,6 +823,22 @@ public void testFieldPositions() {
         }
     }
 
+    @Test
+    public void locale() {
```
Comment (richgillam) [inline icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/number/NumberRangeFormatterTest.java:827]: The name of the test is `locale()`? That seems weird.  Am I missing context in which it makes more sense?

File: icu4j/main/core/src/main/java/com/ibm/icu/number/LocalizedNumberFormatter.java [hunk 2]
```diff
@@ -121,6 +121,16 @@ public Format toFormat() {
         return new LocalizedNumberFormatterAsFormat(this, resolve().loc);
     }
 
+    /**
+     * Disassociate the locale from this formatter.
+     *
+     * @return The fluent chain.
+     * @draft ICU 74
+     */
+    public UnlocalizedNumberFormatter resetLocale() {
```
Comment (richgillam) [inline icu4j/main/core/src/main/java/com/ibm/icu/number/LocalizedNumberFormatter.java:130]: I probably missed the discussion, but why `resetLocale()` in Java and `withoutLocale()` in C++?

**Review by richgillam — COMMENTED** [review]
Generally looks great, but a couple things confused me...

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/number/NumberRangeFormatterTest.java (hunk 1 above)
Comment (sffc) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/number/NumberRangeFormatterTest.java:827]: In the file just above this one, the test is also named `locale()`

File: icu4j/main/core/src/main/java/com/ibm/icu/number/LocalizedNumberFormatter.java (hunk 2 above)
Comment (sffc) [reply icu4j/main/core/src/main/java/com/ibm/icu/number/LocalizedNumberFormatter.java]: I think I simply forgot to change Java! That is why we have code reviews...

**Review by richgillam — APPROVED** [review]
Lookin' good!

**Comment by sffc** [comment]
I'll let this sit for a little while, maybe after the icu 74 release, and then let's not forget to merge it before icu 75 :)

**Comment by sffc** [comment]
Remembered to hit the merge button!

**Pull request merged**